org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/utils/TargetGroupUtils.java

```diff
@@ -11,5 +13,17 @@ public static boolean isContainerInElasticInfrastructure(final ResourceContainer
 						  .stream()
 						  .anyMatch(rc -> rc.getId().equals(container.getId()));
 	}
+
+	public static boolean isContainerInTargetGroup(final ResourceContainer container, 
+												   final TargetGroup targetGroup) {
+		if (targetGroup instanceof final ElasticInfrastructure elasticInfrastructure) {
+			return isContainerInElasticInfrastructure(container, elasticInfrastructure);
+		} else if (targetGroup instanceof final ServiceGroup serviceGroup) {
+			return true; // TODO: How do we check this?
```

Comment:
Via the allocation model, i suppose. 
But neither Assembly nor ResourceEnvironment hold a reference to the allocation :/